modules/mobile-agents/android/client/plugins/IDPProxy/src/org/wso2/mdm/agent/proxy/utils/ServerUtilities.java

```diff
@@ -0,0 +1,364 @@
+		JSONObject params = apiUtilities.getRequestParams();
+		Map<String, String> responseParams = new HashMap<String, String>();
+		
+		switch (httpMethod) {
+			case GET:
+				responseParams = sendGetRequest(url, params, headers);
+				break;
+			case POST:
+				responseParams = sendPostRequest(url, params, headers);
+				break;
+		}
+
+		return responseParams;
+
+	}
+	
+	public static Map<String, String> sendGetRequest(String url, JSONObject params, Map<String, String> headers) throws IDPTokenManagerException{
+		HttpGet httpGet = new HttpGet(url);
+		HttpGet httpGetWithHeaders = (HttpGet) buildHeaders(httpGet, headers, HTTP_METHODS.GET);
+		HttpClient httpClient = getCertifiedHttpClient();
+		Map<String, String> responseParams = new HashMap<String, String>();
+
+		try {
+			HttpResponse response = httpClient.execute(httpGetWithHeaders);
+			responseParams.put(Constants.SERVER_RESPONSE_BODY, getResponseBody(response));
+			responseParams.put(Constants.SERVER_RESPONSE_STATUS,
+			                   String.valueOf(response.getStatusLine().getStatusCode()));
+		} catch (ClientProtocolException e) {
+			throw new IDPTokenManagerException("Invalid client protocol.", e);
+		} catch (IOException e) {
+			responseParams.put(Constants.SERVER_RESPONSE_BODY, "Internal Server Error");
+			responseParams.put(Constants.SERVER_RESPONSE_STATUS,
+			                   Constants.INTERNAL_SERVER_ERROR);
+			throw new IDPTokenManagerException("Server connectivity failure.", e);
+		}
+		
+		return responseParams;
+	}
+	
+	public static Map<String, String> sendPostRequest(String url, JSONObject params, Map<String, String> headers) throws IDPTokenManagerException{
+		HttpPost httpPost = new HttpPost(url);
+		HttpClient httpClient = getCertifiedHttpClient();
+		Map<String, String> responseParams = new HashMap<String, String>();
+		
+		if (params != null) {
+			try {
+				httpPost.setEntity(new StringEntity(params.toString()));
+			} catch (UnsupportedEncodingException e) {
+				throw new IDPTokenManagerException("Invalid encoding type.", e);
+			}
+		}
+
+		HttpPost httpPostWithHeaders = (HttpPost) buildHeaders(httpPost, headers, HTTP_METHODS.POST);
+		try {
+			HttpResponse response = httpClient.execute(httpPostWithHeaders);
+			String status = String.valueOf(response.getStatusLine().getStatusCode());
+
+			responseParams.put(Constants.SERVER_RESPONSE_BODY, getResponseBody(response));
+			responseParams.put(Constants.SERVER_RESPONSE_STATUS, status);
+
+		} catch (ClientProtocolException e) {
+			throw new IDPTokenManagerException("Invalid client protocol.", e);
+		} catch (IOException e) {
+			responseParams.put(Constants.SERVER_RESPONSE_BODY, "Internal Server Error");
+			responseParams.put(Constants.SERVER_RESPONSE_STATUS,
+			                   Constants.INTERNAL_SERVER_ERROR);
+			throw new IDPTokenManagerException("Server connectivity failure.", e);
+		}
+		
+		return responseParams;
+	}
+
+	public static Map<String, String> postDataAPI(EndPointInfo apiUtilities,
```

Comment:
Let's change the variable name "apiUtilities" to something more meaningful such as "endpointInfo".